app/models/dummy_auth_source.rb

```diff
@@ -0,0 +1,39 @@
+class DummyAuthSource < AuthSource
+  def test_connection
+    # the dummy connection is always available
+  end
+
+  def authenticate(login, password)
+    existing_user(login, password) || on_the_fly_user(login)
+  end
+
+  def auth_method_name
+    'DerpLAP'
```

Comment:
DerpLAP, sounds promising. Where can I get me some of this cool new tech? :horse: